artemis-server/src/main/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImpl.java

```diff
@@ -3489,10 +3498,19 @@ private Object trackCollectionOrMap(String name, Object resolved, Object bean) {
 
       private Object findByNameProperty(String key, Collection collection) throws InvocationTargetException, IllegalAccessException, NoSuchMethodException {
          // locate on name property, may be a SimpleString
-         for (Object candidate : collection) {
-            Object candidateName = getProperty(candidate, "name");
-            if (candidateName != null && key.equals(candidateName.toString())) {
-               return candidate;
+         if (isClassProperty(key)) {
```

Comment:
The comment on the line above this should either be adjusted, or move down toward the code it applied to originally.